src/comp.py

```diff
@@ -23,49 +25,49 @@ def __repr__(self):
 # whose name starts with 'D':
 
 print("Starts with D:")
-r = []  # TODO
+r = [h.__repr__() for h in humans if h.name[0] is 'D']  # TODO
 print(r)
 
 # Write a list comprehension that creates a list of names of everyone
 # whose name ends in "e".
 
 print("Ends with e:")
-r = []  # TODO
+r = [h.__repr__() for h in humans if h.name[-1] is 'e']  # TODO
 print(r)
 
 # Write a list comprehension that creates a list of names of everyone
 # whose name starts with any letter between 'C' and 'G' inclusive.
 
 print("Starts between C and G, inclusive:")
-r = []  # TODO
+r = [h.__repr__() for h in humans if ord(h.name[0]) in range(67,72) ]  # TODO
 print(r)
 
 # Write a list comprehension that creates a list of all the ages plus 10.
 print("Ages plus 10:")
-r = []  # TODO
+r = [h.age + 10 for h in humans]  # TODO
 print(r)
 
 # Write a list comprehension that creates a list of strings which are the name
 # joined to the age with a hyphen, for example "David-31", for all humans.
 print("Name hyphen age:")
-r = []  # TODO
+r = [h.name + '-' + str(h.age) for h in humans]  # TODO
 print(r)
 
 # Write a list comprehension that creates a list of tuples containing name and
 # age, for example ("David", 31), for everyone between the ages of 27 and 32,
 # inclusive.
 print("Names and ages between 27 and 32:")
-r = []  # TODO
+r = [(h.name,h.age) for h in humans if h.age in range(27,33)]  # TODO
 print(r)
 
 # Write a list comprehension that creates a list of new Humans like the old
 # list, except with all the names capitalized and the ages with 5 added to them.
 # The "humans" list should be unmodified.
 print("All names capitalized:")
-r = []  # TODO
+r = [h.name.upper() +'-'+ str(h.age+5) for h in humans]  # TODO
```

Comment:
The exercise wants us to _create a list of new Humans_, so we have to instantiate a new Human object like so:

```py
[Human(h.name.upper(), h.age + 5) for h in humans]
```